src/components/UI/Controls/Button/index.js

```diff
@@ -134,7 +134,27 @@ const Wrapper = styled.div`
   }
 `;
 
-const Button = (props) => {
+/**
+ * @typedef {object} IButtonProps
+ * @property {'button' | 'submit' | 'reset'} [type]
+ * @property {'primary' | 'danger' | 'default' | 'link' | 'warning'} [bsStyle]
+ * @property {'sm' | 'lg'} [bsSize]
+ * @property {boolean} [disabled]
+ * @property {boolean} [loading]
+ * @property {'left'} [loadingPosition]
+ * @property {React.ReactNode} [children]
+ * @property {string} [className]
+ * @property {number} [loadingTimeout]
+ * @property {string} [href]
+ * @property {'_blank'} [target]
+ * @property {string} [rel]
+ * @property {React.MouseEventHandler<HTMLElement>} [onClick]
+ */
+
+/**
+ * @type {React.ForwardRefExoticComponent<React.PropsWithoutRef<IButtonProps> & React.RefAttributes<HTMLElement>>}
+ */
+const Button = React.forwardRef((props, ref) => {
```

Comment:
Not pretty, but after adding ref forwarding, TypeScript was inferring the wrong type from here, and I couldn't pass any props without getting errors. So until we refactor this into TypeScript, JSDoc does the job.